### PR TITLE
[#211] Fix adapters tests helpers

### PR DESCRIPTION
**Pull request by harenson**

### CODE REVIEW

File: src/adapter_test_helpers/sumo_test_people.erl [hunk 1]
```diff
@@ -46,7 +57,10 @@
 %%% sumo_doc callbacks
 %%%=============================================================================
 
--spec sumo_sleep(person()) -> sumo:doc().
+-spec sumo_schema() -> no_return().
+sumo_schema() -> throw(should_be_implemented_by_children).
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_test_people.erl]: According to **Compiler**:
> Warning: function sumo_schema/0 is unused

Comment (gadgetci) [inline src/adapter_test_helpers/sumo_test_people.erl]: According to **Xref**:
> `sumo_test_people:sumo_schema/0` is an unused local function

Comment (gadgetci) [inline src/adapter_test_helpers/sumo_test_people.erl]: According to **Dialyzer**:
>  Function sumo_schema/0 will never be called

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 2]
```diff
@@ -19,18 +19,18 @@ find_by_sort(Config) ->
...
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
 
-  [First1, Second1 | _] = sumo:find_by(Module, [
-    {age, '>', 2},
-    {age, '=<', 5}
-  ], {age, desc}, 0, 0),
-  "C" = Module:name(First1),
-  "D" = Module:name(Second1),
+  [First1, Second1 | _] = sumo:find_by(Module,
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Dialyzer**:
>  The call sumo:find_by(Module:atom(),[{'age', '>', 2}, {'age', '=<', 5}],{'age', 'desc'},0,0) breaks the contract (DocName,Conditions,SortFields,Limit,Offset) -> Res when is_subtype(DocName,schema_name()), is_subtype(Conditions,conditions()), is_subtype(SortFields,sort()), is_subtype(Limit,non_neg_integer()), is_subtype(Offset,non_neg_integer()), is_subtype(Res,[user_doc()])

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 3]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   {Today, _} = Module:created_at(First),
   % Check that it returns what we have inserted
-  [LastPerson | _NothingElse] = sumo:find_by(Module, [{last_name, "LastName"}]),
-  "Name" = Module:name(LastPerson),
-  "LastName" = Module:last_name(LastPerson),
+  [LastPerson | _NothingElse] = sumo:find_by(Module,
+                                             [{last_name, <<"LastName">>}]),
+  <<"Name">> = Module:name(LastPerson),
+  <<"LastName">> = Module:last_name(LastPerson),
   3 = Module:age(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 63. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 4]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
   {_, Module} = lists:keyfind(module, 1, Config),
 
   [First, Second | _] = sumo:find_all(Module, age, 0, 0),
-  "F" = Module:name(First),
-  "E" = Module:name(Second),
+  <<"F">> = Module:name(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 41. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 5]
```diff
@@ -19,18 +19,18 @@ find_by_sort(Config) ->
 
   [First, Second | _] = sumo:find_by(Module, [{age, '>', 2}], age, 0, 0),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 22. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_conditionals_test_helper.erl [hunk 6]
```diff
@@ -202,9 +188,12 @@ init_store(Module) ->
...
 
-  sumo:persist(Module, Module:new("Jane", "Doe")),
-  sumo:persist(Module, Module:new("John", "Doe", 30)),
-  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5)),
-  sumo:persist(Module, Module:new("Joe", "Armstrong")),
-  sumo:persist(Module, Module:new("Alan", "Turing", 102, "Computer St.")),
+  sumo:persist(Module, Module:new(<<"Jane">>, <<"Doe">>)),
+  sumo:persist(Module, Module:new(<<"John">>, <<"Doe">>, 30)),
+  sumo:persist(Module, Module:new(<<"Jane Jr.">>, <<"Doe">>, 5)),
+  sumo:persist(Module, Module:new(<<"Joe">>, <<"Armstrong">>)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_conditionals_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 194. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 7]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
   3 = Module:age(First),
   4 = Module:age(Second),
-  "D" = Module:last_name(First),
-  "" = Module:address(First),
+  <<"D">> = Module:last_name(First),
+  undefined = Module:address(First),
   Today = Module:birthdate(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 54. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 8]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  [_, _] = Results,
   SortFun = fun(A, B) -> Module:name(A) < Module:name(B) end,
   [First, Second | _] = lists:sort(SortFun, Results),
 
   {Today, _} = calendar:universal_time(),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 49. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 9]
```diff
@@ -95,16 +100,14 @@ delete(Config) ->
   %% delete_by
   2 = sumo:delete_by(Module, [{last_name, <<"D">>}]),
   Results = sumo:find_by(Module, [{last_name, <<"D">>}]),
-
-  0 = length(Results),
+  [] = Results,
 
   %% delete
   [First | _ ] = All = sumo:find_all(Module),
   Id = Module:id(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 107. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 10]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   SortFun = fun(A, B) -> Module:name(A) < Module:name(B) end,
   [First, Second | _] = lists:sort(SortFun, Results),
 
   {Today, _} = calendar:universal_time(),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
   3 = Module:age(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 50. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 11]
```diff
@@ -68,16 +69,20 @@ find_by(Config) ->
   {Today, _} = Module:created_at(LastPerson),
 
   %% Check find_by ID
-  [First1] = sumo:find_by(Module, [{id, Module:id(First)}]),
+  FirstId = Module:id(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 72. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 12]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  undefined = Module:height(First),
+  undefined = Module:description(First),
   {Today, _} = Module:created_at(First),
   % Check that it returns what we have inserted
-  [LastPerson | _NothingElse] = sumo:find_by(Module, [{last_name, "LastName"}]),
-  "Name" = Module:name(LastPerson),
-  "LastName" = Module:last_name(LastPerson),
+  [LastPerson | _NothingElse] = sumo:find_by(Module,
+                                             [{last_name, <<"LastName">>}]),
+  <<"Name">> = Module:name(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 61. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 13]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
   3 = Module:age(First),
   4 = Module:age(Second),
-  "D" = Module:last_name(First),
-  "" = Module:address(First),
+  <<"D">> = Module:last_name(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 52. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 14]
```diff
@@ -142,6 +145,14 @@ init_store(Module) ->
   sumo:persist(Module, Module:new(<<"Model T-2000">>, <<"undefined">>, 7)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 145. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 15]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
...
-  sumo:persist(Module, Module:new("A", "E", 6)),
-  sumo:persist(Module, Module:new("B", "D", 3)),
-  sumo:persist(Module, Module:new("C", "C", 5)),
-  sumo:persist(Module, Module:new("D", "B", 4)),
-  sumo:persist(Module, Module:new("E", "A", 2)),
-  sumo:persist(Module, Module:new("F", "E", 1)).
+  sumo:persist(Module, Module:new(<<"A">>, <<"E">>, 6)),
+  sumo:persist(Module, Module:new(<<"B">>, <<"D">>, 3)),
+  sumo:persist(Module, Module:new(<<"C">>, <<"C">>, 5)),
+  sumo:persist(Module, Module:new(<<"D">>, <<"B">>, 4)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 64. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 16]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
   {_, Module} = lists:keyfind(module, 1, Config),
 
   Results = sumo:find_by(Module, [{last_name, <<"D">>}]),
-  2 = length(Results),
+  [_, _] = Results,
   SortFun = fun(A, B) -> Module:name(A) < Module:name(B) end,
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 43. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 17]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
   sumo:create_schema(Module),
   sumo:delete_all(Module),
 
-  sumo:persist(Module, Module:new("A", "E", 6)),
-  sumo:persist(Module, Module:new("B", "D", 3)),
-  sumo:persist(Module, Module:new("C", "C", 5)),
-  sumo:persist(Module, Module:new("D", "B", 4)),
-  sumo:persist(Module, Module:new("E", "A", 2)),
-  sumo:persist(Module, Module:new("F", "E", 1)).
+  sumo:persist(Module, Module:new(<<"A">>, <<"E">>, 6)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 61. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_conditionals_test_helper.erl [hunk 18]
```diff
@@ -202,9 +188,12 @@ init_store(Module) ->
...
+  sumo:persist(Module, Module:new(<<"Alan">>,
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_conditionals_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 195. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 19]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
...
-    {last_name, desc},
-    {age, asc}
-  ], 0, 0),
-  "F" = Module:name(First1),
-  "A" = Module:name(Second1),
+  [First1, Second1 | _] = sumo:find_all(Module,
+                                        [{last_name, desc}, {age, asc}],
+                                        0,
+                                        0),
+  <<"F">> = Module:name(First1),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 48. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 20]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
...
+  <<"A">> = Module:name(Second1),
 
   [First2, Second2 | _] = sumo:find_all(Module, last_name, 0, 0),
-  "E" = Module:name(First2),
-  "D" = Module:name(Second2),
+  <<"E">> = Module:name(First2),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 52. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 21]
```diff
@@ -19,18 +19,18 @@ find_by_sort(Config) ->
...
-    {age, '=<', 5}
-  ], {age, desc}, 0, 0),
-  "C" = Module:name(First1),
-  "D" = Module:name(Second1),
+  [First1, Second1 | _] = sumo:find_by(Module,
+                                       [{age, '>', 2}, {age, '=<', 5}],
+                                       {age, desc},
+                                       0,
+                                       0),
+  <<"C">> = Module:name(First1),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 30. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 22]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  [LastPerson | _NothingElse] = sumo:find_by(Module,
+                                             [{last_name, <<"LastName">>}]),
+  <<"Name">> = Module:name(LastPerson),
+  <<"LastName">> = Module:last_name(LastPerson),
   3 = Module:age(LastPerson),
-  "" = Module:address(LastPerson),
+  undefined = Module:address(LastPerson),
   {Date, _} = calendar:universal_time(),
   Date = Module:birthdate(LastPerson),
   1.75 = Module:height(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 67. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 23]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
...
 
-  sumo:persist(Module, Module:new("A", "E", 6)),
-  sumo:persist(Module, Module:new("B", "D", 3)),
-  sumo:persist(Module, Module:new("C", "C", 5)),
-  sumo:persist(Module, Module:new("D", "B", 4)),
-  sumo:persist(Module, Module:new("E", "A", 2)),
-  sumo:persist(Module, Module:new("F", "E", 1)).
+  sumo:persist(Module, Module:new(<<"A">>, <<"E">>, 6)),
+  sumo:persist(Module, Module:new(<<"B">>, <<"D">>, 3)),
+  sumo:persist(Module, Module:new(<<"C">>, <<"C">>, 5)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 63. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 24]
```diff
@@ -142,6 +145,14 @@ init_store(Module) ->
   sumo:persist(Module, Module:new(<<"Model T-2000">>, <<"undefined">>, 7)),
 
   {Date, _} = calendar:universal_time(),
-  sumo:persist(Module, Module:new(
-    "Name", "LastName", 3, "", Date, 1.75, <<"description">>)),
+  sumo:persist(
+    Module,
+    Module:new(<<"Name">>,
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 150. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 25]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
...
+  sumo:persist(Module, Module:new(<<"D">>, <<"B">>, 4)),
+  sumo:persist(Module, Module:new(<<"E">>, <<"A">>, 2)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 65. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_conditionals_test_helper.erl [hunk 26]
```diff
@@ -202,9 +188,12 @@ init_store(Module) ->
   sumo:create_schema(Module),
   sumo:delete_all(Module),
 
-  sumo:persist(Module, Module:new("Jane", "Doe")),
-  sumo:persist(Module, Module:new("John", "Doe", 30)),
-  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5)),
-  sumo:persist(Module, Module:new("Joe", "Armstrong")),
-  sumo:persist(Module, Module:new("Alan", "Turing", 102, "Computer St.")),
+  sumo:persist(Module, Module:new(<<"Jane">>, <<"Doe">>)),
+  sumo:persist(Module, Module:new(<<"John">>, <<"Doe">>, 30)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_conditionals_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 192. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 27]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   [First, Second | _] = lists:sort(SortFun, Results),
 
   {Today, _} = calendar:universal_time(),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
   3 = Module:age(First),
   4 = Module:age(Second),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 51. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 28]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
-  2 = length(Results),
+  [_, _] = Results,
   SortFun = fun(A, B) -> Module:name(A) < Module:name(B) end,
   [First, Second | _] = lists:sort(SortFun, Results),
 
   {Today, _} = calendar:universal_time(),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 48. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 29]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  <<"D">> = Module:name(Second),
   3 = Module:age(First),
   4 = Module:age(Second),
-  "D" = Module:last_name(First),
-  "" = Module:address(First),
+  <<"D">> = Module:last_name(First),
+  undefined = Module:address(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 53. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 30]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  undefined = Module:description(First),
   {Today, _} = Module:created_at(First),
   % Check that it returns what we have inserted
-  [LastPerson | _NothingElse] = sumo:find_by(Module, [{last_name, "LastName"}]),
-  "Name" = Module:name(LastPerson),
-  "LastName" = Module:last_name(LastPerson),
+  [LastPerson | _NothingElse] = sumo:find_by(Module,
+                                             [{last_name, <<"LastName">>}]),
+  <<"Name">> = Module:name(LastPerson),
+  <<"LastName">> = Module:last_name(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 62. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 31]
```diff
@@ -68,16 +69,20 @@ find_by(Config) ->
   {Today, _} = Module:created_at(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 69. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 32]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
   {_, Module} = lists:keyfind(module, 1, Config),
 
   [First, Second | _] = sumo:find_all(Module, age, 0, 0),
-  "F" = Module:name(First),
-  "E" = Module:name(Second),
+  <<"F">> = Module:name(First),
+  <<"E">> = Module:name(Second),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 42. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 33]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
...
   sumo:delete_all(Module),
 
-  sumo:persist(Module, Module:new("A", "E", 6)),
-  sumo:persist(Module, Module:new("B", "D", 3)),
-  sumo:persist(Module, Module:new("C", "C", 5)),
-  sumo:persist(Module, Module:new("D", "B", 4)),
-  sumo:persist(Module, Module:new("E", "A", 2)),
-  sumo:persist(Module, Module:new("F", "E", 1)).
+  sumo:persist(Module, Module:new(<<"A">>, <<"E">>, 6)),
+  sumo:persist(Module, Module:new(<<"B">>, <<"D">>, 3)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 62. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 34]
```diff
@@ -58,9 +58,9 @@ init_store(Module) ->
...
+  sumo:persist(Module, Module:new(<<"C">>, <<"C">>, 5)),
+  sumo:persist(Module, Module:new(<<"D">>, <<"B">>, 4)),
+  sumo:persist(Module, Module:new(<<"E">>, <<"A">>, 2)),
+  sumo:persist(Module, Module:new(<<"F">>, <<"E">>, 1)).
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 66. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 35]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   4 = Module:age(Second),
-  "D" = Module:last_name(First),
-  "" = Module:address(First),
+  <<"D">> = Module:last_name(First),
+  undefined = Module:address(First),
   Today = Module:birthdate(First),
-  0.0 = Module:height(First),
-  <<>> = Module:description(First),
+  undefined = Module:height(First),
+  undefined = Module:description(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 56. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 36]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
-  [LastPerson | _NothingElse] = sumo:find_by(Module, [{last_name, "LastName"}]),
-  "Name" = Module:name(LastPerson),
-  "LastName" = Module:last_name(LastPerson),
+  [LastPerson | _NothingElse] = sumo:find_by(Module,
+                                             [{last_name, <<"LastName">>}]),
+  <<"Name">> = Module:name(LastPerson),
+  <<"LastName">> = Module:last_name(LastPerson),
   3 = Module:age(LastPerson),
-  "" = Module:address(LastPerson),
+  undefined = Module:address(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 64. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 37]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
...
+                                        0,
+                                        0),
+  <<"F">> = Module:name(First1),
+  <<"A">> = Module:name(Second1),
 
   [First2, Second2 | _] = sumo:find_all(Module, last_name, 0, 0),
-  "E" = Module:name(First2),
-  "D" = Module:name(Second2),
+  <<"E">> = Module:name(First2),
+  <<"D">> = Module:name(Second2),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 53. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 38]
```diff
@@ -19,18 +19,18 @@ find_by_sort(Config) ->
...
-  ], {age, desc}, 0, 0),
-  "C" = Module:name(First1),
-  "D" = Module:name(Second1),
+  [First1, Second1 | _] = sumo:find_by(Module,
+                                       [{age, '>', 2}, {age, '=<', 5}],
+                                       {age, desc},
+                                       0,
+                                       0),
+  <<"C">> = Module:name(First1),
+  <<"D">> = Module:name(Second1),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 31. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 39]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   {Date, _} = calendar:universal_time(),
   Date = Module:birthdate(LastPerson),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 66. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_conditionals_test_helper.erl [hunk 40]
```diff
@@ -202,9 +188,12 @@ init_store(Module) ->
...
+  sumo:persist(Module, Module:new(<<"Jane Jr.">>, <<"Doe">>, 5)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_conditionals_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 193. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 41]
```diff
@@ -19,18 +19,18 @@ find_by_sort(Config) ->
 
   [First, Second | _] = sumo:find_by(Module, [{age, '>', 2}], age, 0, 0),
 
-  "B" = Module:name(First),
-  "D" = Module:name(Second),
+  <<"B">> = Module:name(First),
+  <<"D">> = Module:name(Second),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 23. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl (hunk 16 above)
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 43. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_conditionals_test_helper.erl [hunk 42]
```diff
@@ -202,9 +188,12 @@ init_store(Module) ->
   sumo:create_schema(Module),
   sumo:delete_all(Module),
 
-  sumo:persist(Module, Module:new("Jane", "Doe")),
-  sumo:persist(Module, Module:new("John", "Doe", 30)),
-  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5)),
-  sumo:persist(Module, Module:new("Joe", "Armstrong")),
-  sumo:persist(Module, Module:new("Alan", "Turing", 102, "Computer St.")),
+  sumo:persist(Module, Module:new(<<"Jane">>, <<"Doe">>)),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_conditionals_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 191. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 43]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
   3 = Module:age(First),
   4 = Module:age(Second),
-  "D" = Module:last_name(First),
-  "" = Module:address(First),
+  <<"D">> = Module:last_name(First),
+  undefined = Module:address(First),
   Today = Module:birthdate(First),
-  0.0 = Module:height(First),
-  <<>> = Module:description(First),
+  undefined = Module:height(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 55. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_basic_test_helper.erl [hunk 44]
```diff
@@ -39,28 +39,29 @@ find_by(Config) ->
...
+  undefined = Module:description(First),
   {Today, _} = Module:created_at(First),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_basic_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 57. Only modules that define callbacks should make dynamic calls.

File: src/adapter_test_helpers/sumo_find_test_helper.erl [hunk 45]
```diff
@@ -38,19 +38,19 @@ find_all_sort( Config) ->
...
-    {age, asc}
-  ], 0, 0),
-  "F" = Module:name(First1),
-  "A" = Module:name(Second1),
+  [First1, Second1 | _] = sumo:find_all(Module,
+                                        [{last_name, desc}, {age, asc}],
+                                        0,
+                                        0),
+  <<"F">> = Module:name(First1),
+  <<"A">> = Module:name(Second1),
```
Comment (gadgetci) [inline src/adapter_test_helpers/sumo_find_test_helper.erl]: According to **Elvis**:
> Remove the dynamic function call on line 49. Only modules that define callbacks should make dynamic calls.

File: Makefile [hunk 46]
```diff
@@ -10,10 +10,11 @@ dep_lager       = git https://github.com/basho/lager.git       3.0.1
 dep_worker_pool = git https://github.com/inaka/worker_pool.git 1.0.4
 dep_uuid        = git https://github.com/okeuday/uuid.git      31f408f4ef
 
-TEST_DEPS = mixer
+TEST_DEPS = katana mixer
+dep_katana = git https://github.com/inaka/erlang-katana.git 0.2.22
```
Comment (elbrujohalcon) [inline Makefile]: Why are you not using the version from hex.pm ?

File: elvis.config [hunk 47]
```diff
@@ -3,7 +3,7 @@
    elvis,
    [
     {config,
-     [#{dirs => ["src", "test"],
+     [#{dirs => ["src/*", "test"],
```
Comment (elbrujohalcon) [inline elvis.config]: I think you need to add `src/*/*` here.

File: elvis.config [hunk 48]
```diff
@@ -22,7 +22,11 @@
                       sumo_find_SUITE,
                       sumo_basic_SUITE,
                       sumo,
-                      sumo_internal]}
+                      sumo_internal,
```
Comment (elbrujohalcon) [inline elvis.config]: You might want to remove the `invalid_dynamic_call` rule altogether, it's not a critical rule for this project and you're already ignore a lot of modules.

**Comment by cabol** [comment]
Looks ok to me!

File: elvis.config (hunk 47 above)
Comment (harenson) [inline elvis.config]: It works without the second wildcard.

Comment (elbrujohalcon) [inline elvis.config]: OK

**Pull request merged**